File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02580/p02580_num5_parsed.py
"""
# Get the input values
H, W, M = map(int, input().split())

# Initialize a 2D array to store the number of targets in each row and column
row_count = [0] * H
col_count = [0] * W

# Initialize a set to store the positions of all the targets
targets = set()

# Iterate through each target
for _ in range(M):
    h, w = map(int, input().split())
    h -= 1  # Subtract 1 to convert the 1-based index to 0-based index
    w -= 1
    # Increment the count of targets in the respective row and column
    row_count[h] += 1
    col_count[w] += 1
    # Add the target position to the set
    targets.add((h, w))

# Find the maximum number of targets that can be destroyed
max_destroyed = max(row_count) + max(col_count)

# Check if any of the target positions have the maximum number of destroyed targets
for h, w in targets:
    # If the number of targets in the respective row or column is equal to the maximum, subtract 1
    if row_count[h] + col_count[w] == max_destroyed:
        max_destroyed -= 1

# Print the answer
print(max_destroyed)
"""

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02580/p02580_num5_parsed.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

int main() {
    int H, W, M;
    cin >> H >> W >> M;
    
    vector<int> row_count(H, 0);
    vector<int> col_count(W, 0);
    unordered_set<int> targets;
    
    for (int i = 0; i < M; i++) {
        int h, w;
        cin >> h >> w;
        row_count[h-1]++;
        col_count[w-1]++;
        targets.insert((h-1) * W + (w-1));
    }
    
    int max_target = 0;
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            int count = row_count[i] + col_count[j];
            if (targets.count(i * W + j)) {
                count--;
            }
            max_target = max(max_target, count);
        }
    }
    
    cout << max_target << endl;
    
    return 0;
}
